src/community/smart-data-loader/src/main/java/org/geoserver/smartdataloader/data/SmartDataLoaderDataAccessFactory.java

```diff
@@ -284,9 +285,15 @@ private File saveMappingDocument(String pathname, String filename, Document mapp
     /** Method that allows to get a DataStoreInfo based on a set of parameters. */
     private DataStoreInfo getDataStoreInfo(Map<String, Serializable> params) throws IOException {
         String jdbcDataStoreId = lookup(DATASTORE_METADATA, params, String.class);
-        Catalog c = getGeoServer().getCatalog();
-        DataStoreInfo ds = c.getDataStore(jdbcDataStoreId);
-        return ds;
+        Catalog catalog = getGeoServer().getCatalog();
+        return catalog.getResourcePool().clone(catalog.getDataStore(jdbcDataStoreId), true);
+    }
+
+    private static Serializable resolveValue(Serializable value, GeoServerEnvironment env) {
```

Comment:
Does not seem to be used anywhere?

Reply:
Right, removed it.  Thanks.

Reply:
Holy leftover, batman!

Thanks, removed.